src/test/groovy/com/synacy/gradprogram/spock/exercise/OrderingServiceSpec.groovy

```diff
@@ -0,0 +1,157 @@
+        foodItem
+    }
+
+    def "cartContainsFoodItem should respond with false if cart does not contains item with type food"() {
+        given:
+        List<Item> testListItemWoFood = [gadget, appliance, clothing]
+        def testCart = new Cart(UUID.randomUUID(), testListItemWoFood)
+
+        when:
+        boolean foodItem = orderingService.cartContainsFoodItem(testCart)
+
+        then:
+        !foodItem
+    }
+
+    def "calculateTotalCostOfCart should respond with the total cost of item inside the cart"() {
+        given:
+        List<Item> testListItem = [food, gadget, appliance, clothing]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        double testTotal = orderingService.calculateTotalCostOfCart(testCart)
+
+        then:
+        testTotal == 55
+    }
+
+    def "isCartEligibleForDiscount should respond with true if total cart price is greater than 50.0 and size is greater than 5"() {
+
+        given:
+        List<Item> testListItem = [food, gadget, appliance, clothing, food2, gadget2]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        boolean isEligible = orderingService.isCartEligibleForDiscount(testCart)
+
+        then:
+        isEligible
+    }
+
+    def "isCartEligibleForDiscount should respond with false if cart price is less than 50.0 and cart size is less than 5"() {
+
+        given:
+        List<Item> testListItem = [food, gadget, clothing]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        boolean isEligible = orderingService.isCartEligibleForDiscount(testCart)
+
+        then:
+        !isEligible
+    }
+
+    def "isCartEligibleForDiscount should respond with false if cart price is greater than 50.0 but cart size is less than 5"() {
+
+        given:
+        List<Item> testListItem = [food, gadget, appliance, clothing]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        boolean isEligible = orderingService.isCartEligibleForDiscount(testCart)
+
+        then:
+        !isEligible
+    }
+
+    def "isCartEligibleForDiscount should respond with false if cart price is less than 50.0 but cart size is greater than 5"() {
+
+        given:
+        List<Item> testListItem = [food2, gadget2, clothing2, gadget, clothing, appliance2]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        boolean isEligible = orderingService.isCartEligibleForDiscount(testCart)
+
+        then:
+        !isEligible
+    }
+
+    @Unroll
+    def "ApplyDiscountToCartItems should respond with discounted cost"() {
+        given:
+        List<Item> testListItem = [food, gadget, appliance, clothing, food2, gadget2]
+        def testCart = new Cart(UUID.randomUUID(), testListItem)
+
+        when:
+        orderingService.applyDiscountToCartItems(testCart)
+
+        then:
+        testCart.items.cost << [1.5,1,2,1,0.5,0.5]
```

Comment:
I think this is not a proper assertion. I would suggest using fewer items (at least 2) to make it easier.